etna/models/holt_winters.py

```diff
@@ -316,19 +332,24 @@ def forecast_components(self, df: pd.DataFrame) -> pd.DataFrame:
         model = self._model
         fit_result = self._result
 
-        if fit_result is None or model is None:
+        if fit_result is None or model is None or self._train_freq is None:
             raise ValueError("This model is not fitted!")
 
+        if df["timestamp"].max() <= self._last_train_timestamp:
+            raise NotImplementedError("In-sample prediction decomposition isn't supported by current implementation.")
```

Comment:
Sounds misleading, we actually support in-sample decomposition, but with another method, may you specify this point in the error(everywhere, not only here)

Reply:
Sure, it should be `min` to cover all cases